forte/data/ontology/top.py

```diff
@@ -914,223 +980,412 @@ def __init__(
+        Otherwise, if it computes and sets the box center y coordinate when the
+        box is both associated with a grid and the
+        offset is set.
 
-    Note: all indices are zero-based and counted from top left corner of
-    image.
+        Returns:
+            The row index of the box center in the image array.
+        """
+        if self._cy is not None:
+            return self._cy
+        else:
+            # if cy computation condition is met, then cy is set
+            if self._grid_cy is not None and self._cy_offset is not None:
+                self._cy = self._grid_cy + self._cy_offset
+            else:
+                self._center_condition_check()
+                raise ValueError("cy is not set.")
+        return self._cy
 
-    Args:
-        pack: the container that this ``Box`` will be added to.
-        image_payload_idx: the index of the image payload in the DataPack's
-            image payload list. If it's not set,
-            it defaults to 0 which meaning it will load the first image payload.
-        cy: the row index of the box center in the image array,
-            the unit is one image array entry.
-        cx: the column index of the box center in the image array,
-            the unit is one image array entry.
-        height: the height of the box, the unit is one image array entry.
-        width: the width of the box, the unit is one image array entry.
-    """
+    @property
+    def cx(self) -> int:
+        """
+        The column index of the box center in the image array.
+        It returns the column index of the box center in the image array
+        directly if the box center is set.
+        Otherwise, if it computes and sets the box center x coordinate when the
+        box is both associated with a grid and the
+        offset is set.
 
-    def __init__(
-        self,
-        pack: PackType,
-        cy: int,
-        cx: int,
-        height: int,
-        width: int,
-        image_payload_idx: int = 0,
-    ):
-        # assume Box is associated with Grids
-        super().__init__(pack, image_payload_idx)
-        # center location
-        self._cy = cy
-        self._cx = cx
-        self._height = height
-        self._width = width
+        Returns:
+            The column index of the box center in the image array.
+        """
+        if self._cx is not None:
+            return self._cx
+        else:
+            # if cx computation condition is met, then cx is set
+            if self._grid_cx is not None and self._cx_offset is not None:
+                self._cx = self._grid_cx + self._cx_offset
+            else:
+                self._center_condition_check()
+                raise ValueError("cx is not set.")
+        return self._cx
 
     @property
-    def center(self):
-        return (self._cy, self._cx)
+    def box_center(self) -> Tuple[int, int]:
+        """
+        Get the box center by using the property function cy() and cx().
```

Comment:
don't write implementation details in function docstring, users often won't care (and might get confused).